NebulaWorld/Factory/FactoryManager.cs

```diff
@@ -25,6 +25,9 @@ sealed class ThreadSafe
         public static readonly ToggleSwitch IgnoreBasicBuildConditionChecks = new ToggleSwitch();
         public static readonly ToggleSwitch DoNotAddItemsFromBuildingOnDestruct = new ToggleSwitch();
 
+        public static bool IsFromClient = false; // this is just a temp fix, better use the ToggleSwitch (but its breaking stuff atm and i dont want to fix it now)
```

Comment:
Is this a complex fix? In my experience, nothing is more permanent than a temporary fix, so I'd rather it not get merged in as-is

Reply:
It can lead to a race condition if players place belts at exactly the same time resulting in one of them having their belts snapped to the position of the other players ones.
Good solution would be to use the ToggleSwitch here but it broke other things while testing, so it needs a bit more work.

seems to be kinda rare to trigger it tho.